apis/r/R/utils-arrow.R

```diff
@@ -384,21 +394,59 @@ get_domain_and_extent_dataframe <- function(tbl_schema, ind_col_names,
           ind_max_dom <- arrow_type_range(ind_col_type) - c(0, ind_ext)
         }
 
-        ind_cur_dom <- ind_max_dom
+        requested_slot <- domain[[ind_col_name]]
+        if (is.null(requested_slot)) {
+          ind_cur_dom <- ind_max_dom
+        } else {
+          ind_cur_dom <- requested_slot
+        }
```

Comment:
Make this if/else assignment a bit more R-tastic
```suggestion
        ind_cur_dom <- if (is.null(requested_slot)) {
          ind_max_dom
        } else {
          requested_slot
        }
```